controllers/platformoperator_controller.go

```diff
@@ -62,47 +67,82 @@ func (r *PlatformOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	log.Info("reconciling request", "req", req.NamespacedName)
 	defer log.Info("finished reconciling request", "req", req.NamespacedName)
 
-	// TODO: flesh out status condition management
 	po := &platformv1alpha1.PlatformOperator{}
 	if err := r.Get(ctx, req.NamespacedName, po); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	defer func() {
 		po := po.DeepCopy()
 		po.ObjectMeta.ManagedFields = nil
+
 		if err := r.Status().Patch(ctx, po, client.Apply, client.FieldOwner("platformoperator")); err != nil {
 			log.Error(err, "failed to patch status")
 		}
 	}()
 
-	desiredBundle, err := r.Sourcer.Source(ctx, po)
+	return r.reconcile(ctx, po)
+}
+
+func (r *PlatformOperatorReconciler) reconcile(ctx context.Context, po *platformv1alpha1.PlatformOperator) (ctrl.Result, error) {
+	bd, err := r.ensureDesiredBundleDeployment(ctx, po)
 	if err != nil {
+		reason := platformtypes.ReasonApplyFailed
+		status := metav1.ConditionUnknown
+		if errors.Is(err, ErrSourceFailed) {
+			reason = platformtypes.ReasonSourceFailed
+			status = metav1.ConditionFalse
+		}
 		meta.SetStatusCondition(&po.Status.Conditions, metav1.Condition{
 			Type:    platformtypes.TypeApplied,
-			Status:  metav1.ConditionUnknown,
-			Reason:  platformtypes.ReasonSourceFailed,
+			Status:  status,
+			Reason:  reason,
 			Message: err.Error(),
 		})
 		return ctrl.Result{}, err
 	}
-	if err := r.Applier.Apply(ctx, po, desiredBundle); err != nil {
-		meta.SetStatusCondition(&po.Status.Conditions, metav1.Condition{
-			Type:    platformtypes.TypeApplied,
-			Status:  metav1.ConditionUnknown,
-			Reason:  platformtypes.ReasonApplyFailed,
-			Message: err.Error(),
-		})
-		return ctrl.Result{}, err
+
+	// check whether the generated BundleDeployment are reporting any
+	// failures when attempting to unpack the configured registry+v1
+	// bundle contents, or persisting those unpack contents to the cluster.
+	if failureCond := util.InspectBundleDeployment(ctx, bd.Status.Conditions); failureCond != nil {
+		meta.SetStatusCondition(&po.Status.Conditions, *failureCond)
+		return ctrl.Result{}, nil
 	}
 	meta.SetStatusCondition(&po.Status.Conditions, metav1.Condition{
 		Type:    platformtypes.TypeApplied,
 		Status:  metav1.ConditionTrue,
 		Reason:  platformtypes.ReasonApplySuccessful,
-		Message: "Successfully applied the desired olm.bundle content",
+		Message: fmt.Sprintf("Successfully applied the %s BundleDeployment resource", bd.GetName()),
 	})
+	platformtypes.SetActiveBundleDeployment(po, bd.GetName())
```

Comment:
I forgot we had this field on the PO status. Could this play a role in gating the sourcing logic?